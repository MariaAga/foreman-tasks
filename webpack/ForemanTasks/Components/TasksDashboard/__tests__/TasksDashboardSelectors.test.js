import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import {
  selectTasksDashboard,
  selectTime,
  selectQuery,
  selectTasksSummary,
  calcStoppedOther,
} from '../TasksDashboardSelectors';

const stoppedResult = {
  error: {
    total: 9,
    recent: 1,
  },
  warning: {
    total: 8,
    recent: 2,
  },
  success: {
    total: 7,
    recent: 3,
  },
  cancelled: {
    total: 5,
    recent: 3,
  },
  pending: {
    total: 11,
    recent: 4,
  },
};

const state = {
  foremanTasks: {
    tasksDashboard: {
      time: 'some-time',
      query: 'some-query',
      tasksSummary: {
        running: {
          recent: 3,
          total: 8,
        },
        paused: {
          recent: 2,
          total: 9,
        },
        stopped: {
          by_result: stoppedResult,
        },
        scheduled: {
          total: 6,
        },
      },
    },
  },
};

const fixtures = {
  'should select tasks-dashboard': () => selectTasksDashboard(state),
  'should select tasks-dashboard when state is empty': () =>
    selectTasksDashboard({}),
  'should select time': () => selectTime(state),
  'should select time when state is empty': () => selectTime({}),
  'should select query': () => selectQuery(state),
  'should select query when state is empty': () => selectQuery({}),
  'should select tasks-summary': () => selectTasksSummary(state),
  'should select tasks-summary when state is empty': () =>
    selectTasksSummary({}),
  'should calcStoppedOther': () => calcStoppedOther(stoppedResult),
};

describe('TasksDashboard - Selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
