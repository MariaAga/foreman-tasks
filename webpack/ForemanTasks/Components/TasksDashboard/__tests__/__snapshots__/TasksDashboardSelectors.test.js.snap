// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksDashboard - Selectors should select query 1`] = `"some-query"`;

exports[`TasksDashboard - Selectors should select query when state is empty 1`] = `Object {}`;

exports[`TasksDashboard - Selectors should select tasks-dashboard 1`] = `
Object {
  "query": "some-query",
  "tasksSummary": Object {
    "paused": Object {
      "recent": 2,
      "total": 9,
    },
    "running": Object {
      "recent": 3,
      "total": 8,
    },
    "scheduled": Object {
      "total": 6,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 1,
          "total": 9,
        },
        "success": Object {
          "recent": 3,
          "total": 7,
        },
        "warning": Object {
          "recent": 2,
          "total": 8,
        },
      },
    },
  },
  "time": "some-time",
}
`;

exports[`TasksDashboard - Selectors should select tasks-dashboard when state is empty 1`] = `Object {}`;

exports[`TasksDashboard - Selectors should select tasks-summary 1`] = `
Object {
  "paused": Object {
    "last": 2,
    "older": 7,
  },
  "running": Object {
    "last": 3,
    "older": 5,
  },
  "scheduled": 6,
  "stopped": Object {
    "error": Object {
      "last": 1,
      "total": 9,
    },
    "success": Object {
      "last": 3,
      "total": 7,
    },
    "warning": Object {
      "last": 2,
      "total": 8,
    },
  },
}
`;

exports[`TasksDashboard - Selectors should select tasks-summary when state is empty 1`] = `
Object {
  "paused": Object {
    "last": 0,
    "older": 0,
  },
  "running": Object {
    "last": 0,
    "older": 0,
  },
  "scheduled": 0,
  "stopped": Object {
    "error": Object {
      "last": 0,
      "total": 0,
    },
    "success": Object {
      "last": 0,
      "total": 0,
    },
    "warning": Object {
      "last": 0,
      "total": 0,
    },
  },
}
`;

exports[`TasksDashboard - Selectors should select time 1`] = `"some-time"`;

exports[`TasksDashboard - Selectors should select time when state is empty 1`] = `"H24"`;
